doc/developer/platform/reconciliation.md

```diff
@@ -0,0 +1,64 @@
+  For example, `show databases` reads from a table that is unconditionally initialized with `materialize`.
+  The reconciler has no good mechanism to distinguish initialization updates from others.
+  * *Current solution:* The coordinator truncates all ephemeral tables on startup using a new instruction.
+  * We could maintain a trace of all inserts and block already-inserted data.
+    This would require us to maintain a potentially long history of inserts.
+  * We cannot clear tables by advancing their compaction frontier to `[]`.
+    This would invalidate all dataflows that depend on the table.
+  * The timestamp of inserts during initialization is the current time, not 0, so we can't use this to distinguish initialization inserts.
+* The controller assumes that dataflows are rendered with a specific initial time.
+  When restarting, we do not render previously rendered dataflows, which might violate the implicit contract expected by the coordinator.
+  The reconciler tries to bring the coordinator up-to-date by replying with a correcting `FrontiersUppers` or `TimestampBindings` response.
+  There might be a brief moment in time when the two are not synchronized and the coordinator could produce invalid commands.
+* Peeks and responses are not handled correctly.
+  When reconnecting while a peek was active, we might send a response to a coordinator that doesn't know about the peek.
+  Currently, the COMPUTE controller discards peek responses it cannot match to pending peeks.
+* We need to validate if STORAGE's timestamp bindings are save to be stored in memory within COMPUTE until a coordinator picks them up.
+* The commands supplied by the COMPUTE controller should be idempotent, but in fact are not.
+  For example, when reading a Kafka topic, we create a new topic name each time we instantiate the source.
+  This is incorrect because we bind an existing identifier to a new definition.
+
+### State
+
+We assume that the mapping of `GlobalId` to an object is unique, i.e., after restarts they still name the same object.
+This allows us to maintain a small amount of state to enable command reconciling.
+
+* For each installed identifier, we maintain its upper frontier.
+
+## Open questions
+
+* The coordinator does not indicate the set of known IDs.
+  This makes it difficult to determine which objects to uninstall once a coordinator reconnects.
+  * The IDs are partially-ordered, which would allow the coordinator to indicate the lower bound of not-in-use IDs.
+* How to handle inserts at a time <= the current object's frontier?
+  We could just discard the data as it would probably be the same information already inserted earlier.
+* When reconnecting, we have to get the coordinator's state up-to-date.
+  The coordinator starts with a frontier of `[0]` when creating objects.
+  STORAGE and coordinator should agree on a frontier before telling COMPUTE about it.
+
+## Response reconciliation
```

Comment:
I should probably write some text here, huh? Lmk when a good time for that is, probably after this lands so that we aren't fighting over this filename.